src/appleseed/renderer/modeling/edf/edf.cpp

```diff
@@ -128,4 +132,62 @@ void* EDF::evaluate_inputs(
     return data;
 }
 
+float EDF::get_max_contribution_scalar(const Source* source) const
+{
+    assert(source);
+
+    if (!source || !source->is_uniform())
+        return numeric_limits<float>::max();
+
+    float value;
+
+    source->evaluate_uniform(value);
+
+    return value;
+}
+
+float EDF::get_max_contribution_spectrum(const Source* source) const
+{
+    assert(source);
+
+    if (!source || !source->is_uniform())
+        return numeric_limits<float>::max();
+
+    Spectrum spectrum;
+
+    source->evaluate_uniform(spectrum);
+
+    return max_value(spectrum);
+}
+
+float EDF::get_max_contribution(const Source* input) const
```

Comment:
I added it because in the `constantenvironmentedf.cpp` file, I saw that non zero emission is checked using only the spectrum which is like this: `check_non_zero_emission("radiance");`. I will delete it if you think we will never have such case.